cesium/internal/domain/index.go

```diff
@@ -78,20 +101,50 @@ func (idx *index) overlap(tr telem.TimeRange) bool {
 	return overlap
 }
 
-func (idx *index) update(ctx context.Context, p pointer) (err error) {
+func (idx *index) update(ctx context.Context, p pointer, persist bool) error {
 	_, span := idx.T.Bench(ctx, "update")
-	idx.mu.RLock()
+	idx.mu.Lock()
+
+	defer span.End()
+
 	if len(idx.mu.pointers) == 0 {
-		idx.mu.RUnlock()
-		return span.EndWith(RangeNotFound)
+		idx.mu.Unlock()
+		return span.Error(RangeNotFound)
```

Comment:
do you want a test that covers the case of the condition on 110 to be true?  maybe codecov is just not up to date and the tests you have above test it already but just wanted to make sure

Reply:
I think this case never happens: `update` is only called in `Commit`, and `Commit` would only call `update` if prevCommit is not 0, which means the `len(pointers) != 0`. This is just here as a safety, even though it never happens. Should this still be in the code?

Reply:
If that is the correct way to handle events that should never happen in go (i just am not convinced it is yet), then it would be fine. In C++ I'd place an assertion for things I'm positive would not happen but not sure if the same paradigm is used here (stack suggests panic would be the equivalent way to handle). In any case, make sure its clear in the code at least with a comment that this case should never happen

Reply:
@emilbon99 What should happen here? Should panic?